File: Python/Program/Villager.py
from Character import Character
from Store import Store

class Villager(Character):
    def __init__(self, name: str, lives_at: str, main_skill: str, likes: list, occupation: str, store: Store):
        super().__init__(name, lives_at, main_skill)
        self._likes = likes
        self._occupation = occupation
        self._store = store

    def get_likes(self) -> list:
        return self._likes

    def set_likes(self, likes: list):
        self._likes = likes

    def get_occupation(self) -> str:
        return self._occupation

    def set_occupation(self, occupation: str):
        self._occupation = occupation

    def get_store(self) -> Store:
        return self._store

    def set_store(self, store: Store):
        self._store = store

File: C++/Program/Villager.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Character.cpp"
#include "Store.cpp"

using namespace std;

class Villager : public Character {
private:
    vector<string> likes;
    string occupation;
    Store store;

public:
    // Constructor
    Villager(){}

    Villager(string name, string livesAt, string mainSkill, vector<string> likes, string occupation, Store store)
        : Character(name, livesAt, mainSkill) {
        this->likes = likes;
        this->occupation = occupation;
        this->store = store;
    }

    // Getter and Setter for likes
    vector<string> getLikes() {
        return likes;
    }

    void setLikes(vector<string> likes) {
        this->likes = likes;
    }

    // Getter and Setter for occupation
    string getOccupation() {
        return occupation;
    }

    void setOccupation(string occupation) {
        this->occupation = occupation;
    }

    // Getter and Setter for store
    Store getStore() {
        return store;
    }

    void setStore(Store store) {
        this->store = store;
    }
};